### ISSUE TITLE
Submission: Group 6: Maternal Health Classification

### ISSUE THREAD

**Issue by JiaQuanLim** [UBC-MDS/data-analysis-review-2024#18]
**Submitting authors:**  @hovhannisyannelli @Shan-non @JiaQuanLim

**Repository:** https://github.com/UBC-MDS/maternal_health_classification
**Report link:** https://github.com/UBC-MDS/maternal_health_classification/blob/main/reports/maternal_health_classification.pdf
**Abstract/executive summary:** In this project we are comparing multiple classification models to predict pregnant women’s maternal health risk as low, medium or high from their health data. Maternal health broadly refers to overall health of women during their pregnancy, child birth and their post-natal period (World Health Organization 2024). A variety of complications can arise during pregnancy, childbirth and soon after that result in maternal death. The World Health Organization defines maternal mortality as “the death of a woman whilst pregnant or within 42 days of delivery or termination of pregnancy, from any cause related to, or aggravated by pregnancy or its management, but excluding deaths from incidental or accidental causes” (World Health Organization 1992). Thus, maternal health risk refers to the approximate risk level of a woman’s health while pregnant or soon after birth. With our chosen model we aim to identify some key indicators that predict higher maternal health risk. In rural communities where it is costly and diﬀicult to provide consistent medical care, having a method to predict maternal health risk from minimally invasive methods could be greatly beneficial in improving health outcomes for mothers and babies alike (Ahmed et al. 2020). After some initial exploration of classification models we settled on the Decision Tree algorithm for its easily interpretable model and relatively high accuracy score. Based on this Decision Tree model a key indicator of increased maternal health risk was blood sugar. The decision tree our model built based on the dataset suggests that blood sugar higher than 7.95 mmol/l is correlated with a high maternal health risk. However, the model does seem to struggle with classifying low vs. medium maternal health risk. While this is a great first step the model accuracy is not accurate enough to be used in a medical context yet. More research and modelling (perhaps with different models) should be done before deploying this technology in rural communities. Additionally, it should be noted that since this dataset contains health data only from a Pima Indians Diabetes
Database the model have may learned a bias specific to unknown genetic factors present in this sample. Thus, a much larger dataset with a diverse sample set should be used to train the model before it is utilized in any communities.

**Editor:** @ttimbers 
**Reviewer:** Ren Zu Hui Zoe, Hrayr Muradyan, Francisco Ramirez Saavedra, Siddarth Subrahmanian

- [x] I agree to abide by [MDS's Code of Conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by HrayrMuradyan** [comment]
## Data analysis review checklist

### Reviewer: HrayrMuradyan

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [ ] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 

### Review Comments: 

Overall a good job! Here are some comments that I think should be changed:

**Results:**: Please take a look at _Figure 3: Boxplot of the Maternal Health Dataset_ on page 7. Firstly, the figure is very small in the report and it's hard to see it. Secondly, the title is uninformative. "Boxplot of the Maternal Health Dataset" is too general and does not refer to specific variables in the dataset. After zooming into the plot it can be seen that you plot the boxplots of different numeric variables based on the Risk Level (Target variable). So, it will be more informative if you contain that message in the plot description. This refers to some other plots as well. Overall good job!

**Conclusions**: While you have discussion in the summary of the report, the report misses a discussion section or at least a discussion paragraph at the end. You don't conclude your work by talking about how it can be used. Additionally, you don't talk about the potential limitations and what future work can be done to improve those and improve your project even further. 

**Analysis**: Good job at creating the references! In the Analysis section, however, the exact tools are not mentioned. Instead, the group used the names of the authors instead of the exact tools. For example, instead of mentioning Pandas, McKinney (2010) is only mentioned. It will be better to write "we used Pandas (McKinney 2010)".

**Data Validation**: While the analysis contains lots of valuable checks, I couldn't find the check for duplicate rows. I can't find Feature-Target correlation check either. It is important to be sure there are no duplicate rows (What if one of them goes to test data, another one to train data?). Feature-Target correlation is another important thing to check, what if one of the variables explain the target variable perfectly?

**References**: I would recommend to also include the date the data was published (9 May 1990). It will provide additional insights about how old is the dataset.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by fraramfra** [comment]
## Data analysis review checklist

### Reviewer: @fraramfra - Francisco (Paco) Ramirez Saavedra

### Conflict of interest

- [X] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [X] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [X] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [X] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [X] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [X] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [X] **Style guidelines:** Does the code adhere to well known language style guides?
- [X] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robustness?

### Reproducibility
 
- [X] **Data:** Is the raw data archived somewhere? Is it accessible?
- [X] **Computational methods:** Is all the source code required for the data analysis available?
- [X] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [X] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [X] **Authors:** Does the report include a list of authors with their affiliations?
- [X] **What is the question:** Do the authors clearly state the research question being asked?
- [X] **Importance:** Do the authors clearly state the importance for this research question?
- [X] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [X] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [X] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [X] **Conclusions:** Are the conclusions presented by the authors correct? 
- [X] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [X] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 2.5

### Review Comments:

Good job, guys! 
The general structure makes it easy to follow your storytelling. You were very thorough in testing different models, exploring different alternatives to tackle the problem.
The README file was clear and easy to follow from going into the container, running scripts and obtaining the reports. I was able to download the repo, deploy the container and run the analysis down to getting the reports very easily.

Here are a few identified areas of opportunity for the analysis:

Report:

- In the Analysis section, the references regarding Python and package use are unclear, as no actual packages are mentioned.
- It is unclear what type of validation checks were performed beyond checking for duplicated rows. Could benefit from mentioning briefly how/if Pandera (DataFrame) or DeepChecks were used for data validation.
- Table 3 seems to contain redundant information already contained in Tables 1 and 2 (number of rows and columns are explicit in the other two tables).
- Fig 1 could benefit from being sorted using the ordinal values of the target.
- EDA could benefit from including visualization of the distributions of the different features. Even though Table 1 describes said distributions, a plot could help make it easier to visualize correlations mentioned on the heatmap and find others.
- Figure 3 is too small and difficult to read in the PDF version of the report. However, it has been noted that the sizing is appropriate in the html report.
- Though it was mentioned that the used Decision Tree went through hyperparameter optimization, it would be informative to mention if other models also went through any type of optimization prior to model selection. In case there were not, you could comment on it on your Summary notes regarding future work.
- The report could benefit from an explicit section on Conclusions, even if they are mentioned in the Summary.
- It was noted that you were very cautious in extracting tables and values used in the report in your quizzes instead of hard-coding results in the report. Good job on that! Still, there are a few instances, particularly in your last paragraph of the "Reporting Test Score" where you have some threshold values hard-coded that could be open to error if the analysis is updated in the future.
- Note that the README references are different from those included in the report.

Scripts:

- It is often useful to have print out messages in different steps of the scripts so, in case something goes wrong, it is easier to troubleshoot specific sections of the code if all scripts are run in batch.
- Data download script has a lot of testing and ValueErrors defaults. However, other scripts could benefit from assert tests to ensure functions are working as intended.
- It could be beneficial to have the Pandera validation log created to have a record of examples that did not comply with your checks.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by sgdkd** [comment]
## Data analysis review checklist

### Reviewer: sgdkd

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [ ] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing:  1

### Review Comments: 
**Done particularly well:**
1. The introduction and problem statement are very clear, especially the precise definition of the problem.
2. The exploration of various models (Gaussian Bayes, Logistic Regression, Decision Tree, RBF-SVC) is excellent, clearly demonstrating your selection of the best-performing approach and the reasons behind your choice.
3. The explanation of feature selection and data preprocessing is also well-done and clearly articulated.

**Could be improved:**
1. Since you already have a report link, it would be helpful to add it to the README for easier access.
2. The CONTRIBUTING.md file still shows 'Breast Cancer Predictor project'; this should be updated for consistency.
3. The systolic (upper) and diastolic (lower) blood pressure readings show a strong correlation of 0.78. It may still be necessary to address potential multicollinearity, or use metrics like VIF or Condition Number to confirm this.
4. The section titled 'Report Test Score' is not equivalent to 'Results and Discussion,' especially for a report that may be read by non-experts in data science or medical science. It would be better to interpret your results more formally and make a clear conclusion in your report.
5. Adding a section on potential improvements or future research directions based on your findings would strengthen your work. For example, what limitations did you encounter during the experiment, and what new hypotheses do you have based on your results? This would be particularly useful for scholars who might read or cite your report and build upon your work.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by siddart-1** [comment]
## Data analysis review checklist

### Reviewer: siddart1

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [ ] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1

### Review Comments: 

- The repository is well-organized, and the code is clean, modular, and easy to understand.
- The documentation, particularly in the README, is comprehensive and provides clear instructions on how to set up and run the analysis.
- The background and methodology are well-explained, with clear descriptions of the data, models, and evaluation metrics used.
- Question is well formed and analysis is done methodically.

Areas of Improvement:

- Add report link to README: Since the report is already available, it would be helpful to add a direct link to it in the README file for easier access. This will save time for users looking for the detailed analysis.
- Results interpretation: The section titled 'Report Test Score' does not fully align with the expected 'Results and Discussion' format, especially for an audience that may not have a strong background in data science or medical science. It would be beneficial to interpret your results more formally, provide context for your findings, and draw clearer conclusions for a broader audience.
- Future research directions: Adding a section on potential improvements or future research directions based on your findings would strengthen the work. For example, what limitations did you encounter during the experiment, and what new hypotheses do you have based on your results? This would be particularly useful for scholars who may read or cite your work and use it as a foundation for further study.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.
